### PR TITLE
Logging errors from service

### PR DESCRIPTION
**Pull request by MatthewZMSU**
# Description

Since the ExpressJS server in case of errors sends the error with contexId and pageId, we can log it during crawling.
I implemented logging of the errors in the service middleware.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I have tested it with my local spiders. Everything worked fine.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: examples/spiders/manual_recaptcha.py [hunk 1]
```diff
@@ -62,4 +61,4 @@ def make_screenshot(self, response: PuppeteerScreenshotResponse, **kwargs):
 
     @staticmethod
     def error(failure: Failure):
-        print(f"We are in error function!")
+        print("We are in error function!")
```
Comment (Yatskov) [inline examples/spiders/manual_recaptcha.py]: Do we need this print?

Comment (MatthewZMSU) [reply examples/spiders/manual_recaptcha.py]: It is the way to explicitly show the user that he has failed the request.

Comment (Yatskov) [reply examples/spiders/manual_recaptcha.py]: Then maybe better to use logging for such messages

File: examples/spiders/manual_recaptcha.py [hunk 2]
```diff
@@ -60,6 +59,5 @@ def make_screenshot(self, response: PuppeteerScreenshotResponse, **kwargs):
         with open("imageToSave.png", "wb") as fh:
             fh.write(base64.b64decode(data))
 
-    @staticmethod
-    def error(failure: Failure):
-        print(f"We are in error function!")
+    def error(self, failure: Failure):
+        self.log("We are in error function!")
```
Comment (Yatskov) [inline examples/spiders/manual_recaptcha.py]: Do you want to log this as simple message, or like error with error level and stacktrace or any meta info?

Comment (MatthewZMSU) [reply examples/spiders/manual_recaptcha.py]: This is just a simple message that something is bad with the spider. Nothing more.

Comment (Yatskov) [reply examples/spiders/manual_recaptcha.py]: Sounds like warning

**Pull request merged**